ui/lib/config-ui/addon/components/login-settings/page/details.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import errorMessage from 'vault/utils/error-message';
+
+/**
+ * @module Page::LoginSettingsRuleDetails
+ * Page::LoginSettingsRuleDetails component is used to display rule information.
+ * Shows detailed data, (eg which namespace it applies to, auth type used etc.) on a selected login rule from the custom login settings list.
+ *
+ * @example
+ * ```js
+ * <Page::LoginSettingsRuleDetails @rule={{this.rule}}  />
+ * ```
+ * @param {object} rule - holds login rule data, { backup_auth_types: string[] eg. ['token'], default_auth_type: string "oidc", disable_inheritance: boolean,
+ * name: string "Login rule 1", namespace: string eg "admin/"}
+ * */
+
+export default class LoginSettingsRuleDetails extends Component {
+  @service capabilities;
+  @service('app-router') router;
+  @service store;
+  @service('flash-messages') flashMessages;
```

Comment:
Since the service variable name matches the service file name we don't need to pass anything here
```suggestion
  @service flashMessages;
```